File: maxhanna.client/src/services/datacontracts/planter/user-plant.ts
import { PlantPhoto } from './plant-photo';

export interface UserPlant {
  id: number;
  userId: number;
  name: string;
  species?: string;
  notes?: string;
  location?: string;
  lastWatered?: Date;
  suggestedWaterHours?: number;
  createdAt: Date;
  updatedAt: Date;
  photos?: PlantPhoto[];
}

File: maxhanna.Server/Controllers/DataContracts/Planter/UserPlant.cs
namespace maxhanna.Server.Controllers.DataContracts.Planter
{
    public class UserPlant
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string Name { get; set; }
        public string? Species { get; set; }
        public string? Notes { get; set; }
        public string? Location { get; set; }
        public DateTime? LastWatered { get; set; }
        public int? SuggestedWaterHours { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public List<PlantPhoto>? Photos { get; set; }
    }
}
